packages/gameserver/src/channels.ts

```diff
@@ -392,17 +394,16 @@ const loadGameserver = async (
 
   const isReady = status.state === 'Ready'
   const isNeedingNewServer =
-    !engineStarted &&
     !config.kubernetes.enabled &&
     (status.state === 'Shutdown' ||
       app.instance == null ||
       app.instance.locationId !== locationId ||
       app.instance.channelId !== channelId)
 
   if (isReady || isNeedingNewServer) {
-    engineStarted = true
     await handleInstance(app, status, locationId, channelId, userId)
-    if (sceneId != null) await loadEngine(app, sceneId)
+    if (!engineStarted && sceneId != null) await loadEngine(app, sceneId)
+    engineStarted = true
```

Comment:
engineStarted needs to be set to true before the async loadEngine is called, otherwise there is potential for loading the engine twice

Reply:
Moved engineStarted before 'await loadEngine'